src/infra/StreamResponse.ts

```diff
@@ -0,0 +1,146 @@
+import { EnumOrLiteral, StreamEvent, StreamEventName } from "../models";
+import { RawResponse } from "../models/http/RawResponse";
+import { StreamEventCallback } from "../models/endpoints/stream/StreamEventCallback";
+
+/**
+ * Wrapper class around {@link RawResponse} that provides
+ * an interface for working with Chat's streaming data in
+ * browser and node environment.
```

Comment:
nit, "in both browser and Node environments" otherwise it sounds like they're combined